File: test_data_do_not_alter/javascript/refactor_original4.js
function reportUserActivity(user) {
    console.log(`${user.name} logged in at ${user.loginTime}`);
    const sessionLength = (user.logoutTime - user.loginTime) / 1000;
    console.log(`Session duration: ${sessionLength} seconds`);
    if (user.failedAttempts > 3) {
        console.log("User had multiple failed login attempts");
    }
    const passwordAge = (user.loginTime - user.passwordLastChanged) / (1000 * 60 * 60 * 24);
    if (passwordAge > 90) {
        console.log("User password may be expired");
    }
}

File: test_data_do_not_alter/python/refactor_original4.py
def report_user_activity(user):
    print(f"{user.name} logged in at {user.login_time}")
    session_length = (user.logout_time - user.login_time).seconds
    print(f"Session duration: {session_length} seconds")
    if user.failed_attempts > 3:
        print("User had multiple failed login attempts")
    if (user.password_last_changed - user.login_time).days > 90:
        print("User password may be expired")
